elastic/kibana
# [Stack Monitoring] Query persistent queue size for metricbeat documents

## Summary

Closes https://github.com/elastic/kibana/issues/132868

The Logstash `Persistent Queue Size` query was missing an aggregation for the documents in ecs format resulting in an empty graph.

### Testing
- start a logstash pipeline that'll inevitably generate a backlog, I used a prebuilt binary.
Make sure to set `queue.type: persisted` in `config/logstash.yml` and start it with the following config `./bin/logstash -f logstash.dev.conf`

logstash.dev.conf
```
input {
  java_generator {
    eps => 10
  }
}

filter {
  sleep {
    time => 2
    every => 1
  }
}

output {
  file {
    path => "/dev/null"
  }
}
```

- collect logstash metrics with metricbeat
```
metricbeat.modules:
  - module: logstash
    xpack.enabled: true
    period: 10s
    hosts: [ "localhost:9600" ]
```

- standalone cluster should appear in Stack Monitoring
- navigate to Logstash > Nodes > [only node in the list] > Advanced

<img width="1705" alt="Screenshot 2022-06-16 at 15 08 37" src="https://user-images.githubusercontent.com/5239883/174081626-22300ae2-c650-4fc6-a9be-6c3a620f3230.png">
